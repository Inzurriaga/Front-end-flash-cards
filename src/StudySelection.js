import React, { Component } from 'react';
import './style/Main.scss';
import Study from "./StudyComponent"
import Controls from "./Controls"

class StudySelection extends Component {
    constructor() {
        super();
        this.state = {
            deck: []
        }
    }

    createDeck = () => {
        let catagorys = []
        document.querySelectorAll('input[name="pick-catagory"]:checked').forEach((picked) => {
            catagorys.push(picked.value);
       });
       let hello = catagorys.reduce((acc, catagory) => {
            let newarray = this.props.cards.filter((cardss) => {
                return cardss.catagory === catagory
            })
            return acc.concat(newarray)
       }, [])
       this.setState({ deck: hello })
    }
    render() {
        if(this.state.deck.length <= 1) {
        return (
            <div>
                <h2>select catagories</h2>
                <Controls create={this.createDeck}
                        selection={this.props.selection}/>
            </div>
        )}else {
            return (
            <div>
                <Study cards={this.state.deck}/>
            </div>
            )
        }
    }
}

export default StudySelection;