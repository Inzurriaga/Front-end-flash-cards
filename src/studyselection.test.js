import React from 'react';
import StudySelection from './StudySelection';
import { shallow } from 'enzyme';

describe('the selection component for the study', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <StudySelection />);
  });

  it('should display study selection', () => {
    expect(wrapper).toMatchSnapshot();
  });
});