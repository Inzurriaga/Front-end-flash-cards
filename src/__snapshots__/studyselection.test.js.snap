// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the selection component for the study should display study selection 1`] = `
<div>
  <h2>
    select catagories
  </h2>
  <Controls
    create={[Function]}
  />
</div>
`;
